routing/record/validation.go

```diff
@@ -100,3 +112,18 @@ func CheckRecordSig(r *pb.Record, pk ci.PubKey) error {
 	}
 	return nil
 }
+
+type RecordSorter []*pb.Record
+
+func (rs RecordSorter) Less(i, j int) bool {
+	mv, _ := MoreValid(rs[j], rs[i])
```

Comment:
Look at this random choice a bit like go's map randomizing access. Users should not be publishing to the same name with validity schemes that are not interoperable. Another choice that would discourage users and be safer may be to just have processes delete all records for a name if they find multiple records with differing validity schemes (like anti-particles annihilating...).

The problem with using a timestamp for every record is that timestamps dont mean anything in some contexts. for example, if i use a git-chain style approach, the last record could be a year old and still be the intended (most valid) one. A timestamp here would make you chose the wrong name. we also get into clock skew problems, and so on... People can't even agree on what notion of time we should use, let alone what the clock says... (also timestamps on every record blow up the size of the record space). Now, the good thing about the timestamp is that it at least is deterministic-- so given the same records everyone picks the same one, which is easier to reason about.

the major failure case i see with the randomized approach is this: a faulty (malicious or broken) process continues to serve an old record of a different validity type. it would cause name confusion. (so yeah, random isn't great, but neither are timestamps). 

I guess the tradeoff lies in how much we want users NOT to do this. If we really want them NOT to do it ever, we should just delete them. If we're ok with it, we should use something like integer increments (vector timestamps) or network-time timestamps.